orangecontrib/text/widgets/tests/test_owstatistics.py

```diff
@@ -0,0 +1,439 @@
+            metas=metas,
+            text_features=[text_var],
+        )
+
+    def _set_feature(self, feature_name: str, value: str = ""):
+        """
+        Set statistic which need to be computed by widget. It sets only one
+        statistics.
+
+        Parameters
+        ----------
+        feature_name
+            The name of statistic
+        value
+            If statistic need a value (e.g. prefix) it is passed here.
+        """
+        feature_index = STATISTICS_NAMES.index(feature_name)
+        self.widget.active_rules = [(feature_index, value)]
+        self.widget.adjust_n_rule_rows()
+
+    def _compute_features(self, feature_name: str, value: str = "") -> Corpus:
+        """
+        Send `self.corpus` to widget, set statistic which need bo be computed,
+        run the computation, and return widget output.
+
+        Parameters
+        ----------
+        feature_name
+            The name of the statistic, only one statistic is set
+        value
+            The value if statistic need it.
+
+        Returns
+        -------
+        Resulting corpus.
+        """
+        self.send_signal(self.widget.Inputs.corpus, self.corpus)
+        self.wait_until_finished()
+        self._set_feature(feature_name, value)
+        self.widget.apply()
+        self.wait_until_finished()
+        res = self.get_output(self.widget.Outputs.corpus)
+        self.assertTupleEqual((len(self.corpus), 1), res.X.shape)
+        return res
+
+    def test_send_data(self):
+        """ Test with basic data, and empty data """
+        self.send_signal(self.widget.Inputs.corpus, self.book_data)
+        self.assertEqual(len(self.book_data), len(self.widget.corpus))
+
+        self.send_signal(self.widget.Inputs.corpus, None)
+        self.assertIsNone(self.widget.corpus)
+        self.widget.apply()
+        self.assertIsNone(self.get_output(self.widget.Outputs.corpus))
+
+    def test_words_count(self):
+        """ Test words count statistic """
+        data = self._compute_features("Word count")
+        np.testing.assert_array_equal(data.X.flatten(), [8, 9, 11, 9])
+
+        self.send_signal(self.widget.Inputs.corpus, None)
+        self.assertIsNone(self.get_output(self.widget.Outputs.corpus))
+
+    def test_characters_count(self):
+        """ Test characters count statistic """
+        data = self._compute_features("Character count")
+        np.testing.assert_array_equal(data.X.flatten(), [47, 44, 48, 51])
+
+        self.send_signal(self.widget.Inputs.corpus, None)
+        self.assertIsNone(self.get_output(self.widget.Outputs.corpus))
+
+    def test_n_gram_count(self):
+        """ Test n-grams count statistic """
+        data = self._compute_features("N-gram count")
+        np.testing.assert_array_equal(data.X.flatten(), [10, 12, 13, 12])
+
+        self.send_signal(self.widget.Inputs.corpus, None)
+        self.assertIsNone(self.get_output(self.widget.Outputs.corpus))
+
+    def test_word_density(self):
```

Comment:
It would be nice to rename the test to reflect the actual function now that it was changed.